src/main/java/updater/JarUpdater.java

```diff
@@ -0,0 +1,255 @@
+package updater;
+
+import javafx.application.Application;
+import javafx.application.Platform;
+import javafx.geometry.Insets;
+import javafx.scene.Group;
+import javafx.scene.Scene;
+import javafx.scene.control.Label;
+import javafx.scene.layout.VBox;
+import javafx.stage.Stage;
+
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
+import java.util.HashMap;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+/**
+ * Note: This class will be compiled into a JAR on its own
+ *
+ * Replaces a JAR file and optionally start that JAR
+ *
+ * Options:
+ * --source string path of source JAR file to replace old JAR
+ * --target string path of target JAR file to be replaced
+ * --execute-jar "y" to execute target JAR after replacement
+ */
+public class JarUpdater extends Application {
+    private final ExecutorService pool = Executors.newSingleThreadExecutor();
+
+    private static final int MAX_RETRY = 10;
+    private static final int WAIT_TIME = 2000;
+    private static final String BACKUP_FILENAME_SUFFIX = "_backup";
+
+    @SuppressWarnings("PMD")
+    // PMD - "Consider using varargs for methods or constructors which take an array the last parameter."
+    public static void main(String[] args) {
```

Comment:
What was I meant was using `// PMD` after the opening brace. That way we don't need to suppress warnings.